### PR TITLE
Fix checkboxes

### PR DESCRIPTION
**Pull request by paroxp**
What
----

At the moment we're dealing with two ways of managing the permissions
via the Admin interface. This is confusing especially when it comes down
to Nunjucks and the lack of types...

Converging onto a single way of dealing with it in the typescript side,
should make it easier to follow for people.

We're fixing the bug with checkboxes being disabled, when there aren't
enough of managers in the organisation.

Closes #505

Writing some tests, to be assured that some bugs will not re-occur.

How to review
-------------

- Sanity check
- See if new tests cover enough of edge cases
- ClickOps your way for a while pretending to be a QA

### CODE REVIEW

File: src/components/org-users/permissions.njk [hunk 1]
```diff
@@ -163,46 +175,67 @@
...
-            items: [
-              {
-                value: "1",
-                html: "<span>Space auditor</span>",
-                checked: values.space_roles[space.metadata.guid].auditors
-              }
-            ]
-          }) }}
+          <input type="hidden" value="{{ values.space_roles[space.metadata.guid].auditors.current }}" name="space_roles[{{ space.metadata.guid }}][auditors][current]">
+          <div class="govuk-form-group">
```
Comment (richardTowers) [inline src/components/org-users/permissions.njk]: This bit is repeated almost identically seven times - maybe worth pulling it out into a macro and just passing in the bits that change, so we'd call something like:

```
{{ roleCheckbox("org_roles", "managers", "Org manager", isCheckboxDisabled) }}
...
{{ roleCheckbox("org_roles", "billing_managers", "Billing manager", isCheckboxDisabled) }}
...
{{ roleCheckbox("org_roles", "developers", "Org developer", false) }}
...
{{ roleCheckbox("org_roles", "auditors", "Org auditor", false) }}
...
{{ roleCheckbox("space_roles", "managers", "Space manager", false) }}
...
{{ roleCheckbox("space_roles", "developers", "Space developer", false) }}
...
{{ roleCheckbox("space_roles", "auditors", "Space auditor", false) }}
```

**Comment by richardTowers** [comment]
👍  looks good!
